banyand/liaison/http/server.go

```diff
@@ -61,16 +62,30 @@ type service struct {
 	srv          *http.Server
 	listenAddr   string
 	grpcAddr     string
+	creds        credentials.TransportCredentials
+	keyFile      string
+	certFile     string
+	grpcCert     string
+	tlsEnabled   bool
 }
 
 func (p *service) FlagSet() *run.FlagSet {
 	flagSet := run.NewFlagSet("")
 	flagSet.StringVar(&p.listenAddr, "http-addr", ":17913", "listen addr for http")
 	flagSet.StringVar(&p.grpcAddr, "grpc-addr", "localhost:17912", "the grpc addr")
+	flagSet.StringVarP(&p.certFile, "cert-file", "", "", "the TLS cert file")
+	flagSet.StringVarP(&p.keyFile, "key-file", "", "", "the TLS key file")
+	flagSet.StringVarP(&p.grpcCert, "grpc-cert-file", "", "", "the GRPC Certfile")
+	flagSet.BoolVarP(&p.tlsEnabled, "tlsEnabled", "", false, "the tls enabling option")
 	return flagSet
 }
 
 func (p *service) Validate() error {
+	creds, errTLS := credentials.NewClientTLSFromFile(p.grpcCert, "")
+	if errTLS != nil {
+		return errors.Wrap(errTLS, "failed to load cert and key")
```

Comment:
The error message needs to be more accurate. `"failed to load the grpc cert"` is better.